module/VuFind/src/VuFind/Recommend/Databases.php

```diff
@@ -126,6 +126,14 @@ class Databases implements RecommendInterface, \Laminas\Log\LoggerAwareInterface
      */
     protected $useLibGuides;
 
+    /**
+     * Configuration of whether to match on the alt_names field in LibGuides
+     * in addition to the primary name
+     *
+     * @var bool
+     */
+    protected $useLibGuidesAlternateNames;
```

Comment:
Since this property isn't set until `setConfig` is called, it might be a good idea to give it a default value. Of course, `setConfig` should always be called in normal circumstances, but it might be good to avoid errors if the class is used in a different way in the future. (Same probably goes for a couple of other existing properties).

Reply:
I agree in theory with the point about protecting if setConfig is not called in the future, but I don't like the idea of declaring the same default value (for each property) in two different places.  That seems to create a maintenance issue  if the default changes or we deprecate something, and it makes the code less clear.  Of course declaring a different default value in the property declaration is no better.  The best approach might be declare defaults and then to change setConfig() to only set the values if configured, i.e.  

if (isset($databasesConfig->useQuery) { $this->useQuery = $databasesConfig->useQuery; }

which is verbose, or

$this->useQuery = $databasesConfig->useQuery ?? $this->useQuery;

which is shorter but looks silly.  And neither seems to be the norm in this codebase generally, or in recommendation modules in particular; i.e. no defaults set during property declaration in AbstractSearchObject.  So I agree with the future-proofing problem but I think the alternatives are worse, unless we want to start a much larger conversation?

Reply:
I actually wouldn't mind either of these approaches. As you say, it may also be a good idea to check other Recommendation modules for problems that fit this pattern (it's an easy thing to overlook, and probably has been overlooked elsewhere), but I don't think we have to solve the problem everywhere in order to address it here. Both of those patterns can be found elsewhere in the codebase (particularly in the Search\*\Options classes), though you might not find such precedents in the Recommendation code, as it is generally not very uniform, since each module is designed to be fairly stand-alone, and many come from different developers.

There's also a third potential approach: create a getter for the property, and have it throw an exception if the getter is called before the value has been set.

I think of the three options, I slightly prefer the null coalescing approach because of its brevity, in spite of the fact that as you say, it looks a bit silly. But I could live with any of them!

Reply:
Ok, as you say I did find examples of both approaches I had suggested, even if they look a bit odd.  I went with the null coalescing approach, like you I think it's best.  As long as there is a comment explaining it, which I adapted from Base/Options.